include/mra/misc/convolutiondata.h

```diff
@@ -0,0 +1,139 @@
+#ifndef MRA_CONVOLUTIONDATA_H
+#define MRA_CONVOLUTIONDATA_H
+
+#include "mra/ops/inner.h"
+#include "mra/misc/hash.h"
+#include "mra/misc/misc.h"
+#include "mra/misc/types.h"
+#include "mra/misc/adquad.h"
+#include "mra/misc/platform.h"
+#include "mra/misc/autocorr.h"
+#include "mra/tensor/tensorview.h"
+
+#define MRA_MAX_LX 7
+namespace mra {
+
+  template <typename T, Dimension NDIM>
+  class ConvolutionData {
+
+    public:
+      using rnl_key = std::tuple<Level, Translation>;
+
+    private:
+      size_type K;
+      Level n;
+      int npt;                  // number of quadrature points
+      Translation lx;
+      Tensor<T, 1> quad_x;      // quadrature points
+      Tensor<T, 1> quad_w;      // quadrature weights
+      Tensor<T, 3> autocorr;    // autocorrelation coefficients
+      Tensor<T, 3> c;           // autocorrelation coefficients
+      Tensor<T, 1> coeff;       // coefficients for the convolution
+      Tensor<T, 1> rnlp;        // rnlp coefficients
+      std::map<rnl_key, std::map<rnl_key, Tensor<T, 2>>> matrixmap; // map for storing rnlp matrices
+      std::mutex mapmutex; // mutex for thread safety
+
+
+      void autoc(){
+        detail::autocorr_get(K, autocorr.data());
+
+        auto c_view = c.current_view();
+        c_view(Slice(0, K-1), Slice(0, K-1), Slice(0, 2*K-1)) = autocorr(Slice(0, K-1), Slice(0, K-1), Slice(0, 2*K-1));
+        c_view(Slice(0, K-1), Slice(0, K-1), Slice(2K, 4*K-1)) = autocorr(Slice(0, K-1), Slice(0, K-1), Slice(2*K, 4*K-1));
```

Comment:
@devreal what is a proper way of doing this?

Reply:
Try this one:

```suggestion
        std::array<Slice,NDIM> slices = {Slice(0, K-1), Slice(0, K-1), Slice(0, 2*K-1)};
        c_view(slices) = autocorr(slices);
        slices = {Slice(0, K-1), Slice(0, K-1), Slice(2K, 4*K-1)};
        c_view(slices) = autocorr(slices);
```

Although this is equivalent and shorter:

```suggestion
        std::array<Slice,NDIM> slices = {Slice(0, K-1), Slice(0, K-1), Slice(0, 4*K-1)};
        c_view(slices) = autocorr(slices);
```